### PR TITLE
Added links for some tutorials and organized the How-to alphabetically.

### LINKED ISSUE #2284: Tutorials README doesn't link to all available tutorials
The file https://github.com/allenai/allennlp/blob/master/tutorials/README.md doesn't seem to link to all tutorials, including (but maybe not limited to, didn't look closely) https://github.com/allenai/allennlp/blob/master/tutorials/how_to/create_a_configuration.md

### PR DESCRIPTION
**Pull request by DumbMachine**
Added links for Tutorials  for the following:
    Creating a Configuration File.
    Using a Debugger.
    Training and Using a Transformer ELMo.

Fixes #2284

### CODE REVIEW

File: tutorials/README.md [hunk 1]
```diff
@@ -28,6 +28,10 @@ Our 2018 EMNLP tutorial is available at https://github.com/allenai/writing-code-
 * [Using pre-trained ELMo representations](how_to/elmo.md)
 * [How to train a model with lazy data](how_to/laziness.md)
 * [How to visualize model internals (BETA)](how_to/visualizing_model_internals.md)
+* [Create your own Configuration File](how_to/create_a_configuration.md)
+* [How to use a Debugger](how_to/using_a_debugger.md)
+* [How to Train and use a Transformer ELMo](how_to/training_transformer_elmo.md)
+
```
Comment (nelson-liu) [inline tutorials/README.md]: this newline is extra, would you mind removing it?

Comment (DumbMachine) [reply tutorials/README.md]: Done.

File: tutorials/README.md [hunk 2]
```diff
@@ -24,10 +24,13 @@ Our 2018 EMNLP tutorial is available at https://github.com/allenai/writing-code-
 
 ## How-to
 
-* [Using span representations](how_to/span_representations.md)
-* [Using pre-trained ELMo representations](how_to/elmo.md)
+* [Create your own Configuration File](how_to/create_a_configuration.md)
 * [How to train a model with lazy data](how_to/laziness.md)
+* [How to train and use a Transformer ELMo](how_to/training_transformer_elmo.md)
+* [How to use a Debugger](how_to/using_a_debugger.md)
```
Comment (nelson-liu) [inline tutorials/README.md]: last nitpicks, i think this link title would be better as `How to Debug Your AllenNLP Code`.

File: tutorials/README.md [hunk 3]
```diff
@@ -24,10 +24,13 @@ Our 2018 EMNLP tutorial is available at https://github.com/allenai/writing-code-
 
 ## How-to
 
-* [Using span representations](how_to/span_representations.md)
-* [Using pre-trained ELMo representations](how_to/elmo.md)
+* [Create your own Configuration File](how_to/create_a_configuration.md)
 * [How to train a model with lazy data](how_to/laziness.md)
+* [How to train and use a Transformer ELMo](how_to/training_transformer_elmo.md)
```
Comment (nelson-liu) [inline tutorials/README.md]: `s/Transformer/Transformer-based/`?

File: tutorials/README.md (hunk 2 above)
Comment (DumbMachine) [reply tutorials/README.md]: Done. 😊

File: tutorials/README.md (hunk 3 above)
Comment (nelson-liu) [reply tutorials/README.md]: sorry, i meant to replace `Transformer` with `Transformer-based`

File: tutorials/README.md [hunk 4]
```diff
@@ -24,10 +24,13 @@ Our 2018 EMNLP tutorial is available at https://github.com/allenai/writing-code-
...
+* [How to Debug Your AllenNLP Coder](how_to/using_a_debugger.md)
```
Comment (nelson-liu) [inline tutorials/README.md]: looks like `Coder` should be `Code`?

**Comment by nelson-liu** [comment]
Thanks @DumbMachine , much appreciated!

**Pull request merged**